ReactiveUI/Cocoa/CommonReactiveSource.cs

```diff
@@ -278,30 +234,74 @@ void resetupAll(IReadOnlyList<TSectionInfo> newSectionInfo)
             //
             // TODO: Instead of listening to Changed events and then reseting,
             // we could listen to more specific events and avoid some reloads.
-            disp.Add(sectionChanging.Subscribe(_ => {
+            disp.Add(sectionChanging.ObserveOn(ImmediateScheduler.Instance).Subscribe(_ => {
```

Comment:
Yes, I explained in the PR that `ImmediateScheduler` is required for correctness so I added it explicitly instead of relying on implicit use. Can remove if you prefer.